### PR TITLE
refactor(packages/card): migrate Card to typescript

### PR DESCRIPTION
**Pull request by mohib0306**
#### Summary
- `Card` => migrate component to typescript

### CODE REVIEW

File: packages/components/card/src/card.tsx [hunk 1]
```diff
@@ -1,10 +1,29 @@
-import PropTypes from 'prop-types';
-import React from 'react';
+// import PropTypes from 'prop-types';
```
Comment (adnasa) [inline packages/components/card/src/card.tsx:1]: what about this?

**Comment by mohib0306** [comment]
I see that in some ts-migrated components we have added readme block in the `package.json` i.e. https://github.com/commercetools/ui-kit/blob/9a66387ab6cc5209b98cb52c85945b0d8808e03b/packages/components/constraints/package.json#L37-39. Is that also required? What is it actually?

**Comment by adnasa** [comment]
> Is that also required?

no I would not do that yet.

---

I think it is a pointer for the important when generating the docs.

**Review by adnasa — APPROVED** [review]

File: packages/components/card/src/card.tsx [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
+   */
+  theme: 'light' | 'dark';
+  /**
+   * Pass a custom CSS class, useful to override the styles.
+   * <br>
+   * NOTE: This is not recommended and should only be used for building new components
+   * that require special style adjustments.
+   */
+  className?: string;
+  children?: ReactNode;
```
Comment (mohib0306) [inline packages/components/card/src/card.tsx:22]: These two props are optional, as they were in the original PropTypes definition.

File: packages/components/card/src/card.tsx [hunk 3]
```diff
@@ -1,10 +1,28 @@
...
-const Card = (props) => (
+type TCardProps = {
+  /**
+   * Determines the visual effect of the card. A raised card has a box shadow while a flat card has just a border.
+   */
+  type: 'raised' | 'flat';
+  /**
+   * Determines the background color of the card.
+   */
+  theme: 'light' | 'dark';
```
Comment (mohib0306) [inline packages/components/card/src/card.tsx:14]: No need to mark these as optional, as they have default values defined

**Review by tdeekens — APPROVED** [review]
Nice. Thanks for picking this up!

**Pull request merged**